### PR TITLE
Add typing for File#download()

### PR DESCRIPTION
**Pull request by ThomasdenH**
This adds Typescript types for File#download().

- [x] Tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

### CODE REVIEW

**Comment by ThomasdenH** [comment]
Running `gts fix` solved all CI issues.

**Comment by ThomasdenH** [comment]
Is this under consideration to be merged?

File: src/file.ts [hunk 1]
```diff
@@ -1306,15 +1319,6 @@ class File extends ServiceObject {
     (this.parent as ServiceObject).delete.call(this, options, callback);
   }
 
-  /**
```
Comment (JustinBeckwith) [inline src/file.ts]: Can we keep the jsdoc comments, and move them up with the type definitions?

**Comment by ThomasdenH** [comment]
I believe I have moved all JSDoc now. Do you want me to duplicate their types, considering they are already defined by Typescript?

**Pull request merged**